src/main/java/com/openshift/jenkins/plugins/pipeline/OpenShiftBuilder.java

```diff
@@ -20,165 +22,233 @@
-    protected final String waitTime;
-    
-    
-    // Fields in config.jelly must match the parameter names in the "DataBoundConstructor"
-    @DataBoundConstructor
-    public OpenShiftBuilder(String apiURL, String bldCfg, String namespace, String authToken, String verbose, String commitID, String buildName, String showBuildLogs, String checkForTriggeredDeployments, String waitTime) {
-    	super(apiURL, namespace, authToken, verbose);
-        this.bldCfg = bldCfg;
-        this.commitID = commitID;
-        this.buildName = buildName;
-        this.showBuildLogs = showBuildLogs;
-        this.checkForTriggeredDeployments = checkForTriggeredDeployments;
-        this.waitTime = waitTime;
-    }
-
-    // generically speaking, Jenkins will always pass in non-null field values.  However, as we have periodically
-    // added new fields, jobs created with earlier versions of the plugin get null for the new fields.  Hence, 
-    // we have introduced the generic convention (even for fields that existed in the intial incarnations of the plugin)
-    // of insuring nulls are not returned for field getters
+
+	protected final String bldCfg;
+	protected final String commitID;
+	protected final String buildName;
+	protected final String showBuildLogs;
+	protected final String checkForTriggeredDeployments;
+	protected final String waitTime;
+	protected final String waitUnit;
+
+
+	// Fields in config.jelly must match the parameter names in the "DataBoundConstructor"
+	@DataBoundConstructor
+	public OpenShiftBuilder(String apiURL, String bldCfg, String namespace, String authToken, String verbose, String commitID, String buildName, String showBuildLogs, String checkForTriggeredDeployments, String waitTime, String waitUnit) {
+		super(apiURL, namespace, authToken, verbose);
+		this.bldCfg = bldCfg;
+		this.commitID = commitID;
+		this.buildName = buildName;
+		this.showBuildLogs = showBuildLogs;
+		this.checkForTriggeredDeployments = checkForTriggeredDeployments;
+		this.waitTime = waitTime;
+		this.waitUnit = waitUnit;
+	}
+
+	// generically speaking, Jenkins will always pass in non-null field values.  However, as we have periodically
+	// added new fields, jobs created with earlier versions of the plugin get null for the new fields.  Hence,
+	// we have introduced the generic convention (even for fields that existed in the initial incarnations of the plugin)
+	// of insuring nulls are not returned for field getters
 
 	public String getCommitID() {
 		return commitID;
 	}
-	
+
 	public String getBuildName() {
 		return buildName;
 	}
 
 	public String getShowBuildLogs() {
 		return showBuildLogs;
 	}
-	
+
 	public String getBldCfg() {
 		return bldCfg;
 	}
-	
+
 	public String getCheckForTriggeredDeployments() {
 		return checkForTriggeredDeployments;
 	}
-	
+
+	// NOTE: This is called by the the parameters in config.jelly
 	public String getWaitTime() {
 		return waitTime;
 	}
-	
+
+	public String getWaitUnit() {
```

Comment:
same comment on changing the if's with the BuildVerifier module 

that said, this method should be moved to a default method within IOpenShiftPlugin, where you call the getters for waitUnit and wait, so that you only have to implement once